test/types/xarray.jl

```diff
@@ -0,0 +1,137 @@
+        end
+
+        @testset "ManyToMany" begin
+            T = FakeManyToManyTransform()
+            @test FeatureTransforms.apply(x, T; dims=1) == ones(2, 6)
+            @test FeatureTransforms.apply(x, T; inds=[1, 2]) == ones(2, 2)
+            @test FeatureTransforms.apply(x, T; dims=2, inds=[2, 3]) == ones(2, 4)
+        end
+
+    end
+
+    @testset "apply!" begin
+        T = FakeOneToOneTransform()
+
+        _x = copy(x)
+        FeatureTransforms.apply!(_x, T)
+        @test _x == ones(2, 3)
+
+        # https://github.com/invenia/FeatureTransforms.jl/issues/68
+        @test_broken FeatureTransforms.apply!(copy(x), T; inds=[1, 2])
+    end
+
+    @testset "apply_append" begin
+
+        @testset "OneToOne" begin
+
+            T = FakeOneToOneTransform()
+
+            @test FeatureTransforms.apply_append(x, T; append_dim=1) == vcat(x, ones(2, 3))
+            @test FeatureTransforms.apply_append(x, T; append_dim=2) == hcat(x, ones(2, 3))
+            @test FeatureTransforms.apply_append(x, T; append_dim=3) == cat(x, ones(2, 3), dims=3)
+
+            @test isequal(
+                FeatureTransforms.apply_append(x, T; dims=1, inds=[1], append_dim=1),
+                vcat(x, ones(1, 3))
+            )
+
+            @test isequal(
+                FeatureTransforms.apply_append(x, T; dims=2, inds=[1, 2], append_dim=2),
+                hcat(x, ones(2, 2))
+            )
+        end
+
+        @testset "OneToMany" begin
+
+            T = FakeOneToManyTransform()
+
+            @test_throws DimensionMismatch FeatureTransforms.apply_append(x, T; append_dim=1)
+            @test_throws DimensionMismatch FeatureTransforms.apply_append(x, T; append_dim=3)
+
+            @test isequal(
+                FeatureTransforms.apply_append(x, T; append_dim=2),
+                hcat(x, ones(2, 6))
+            )
+
+            @test isequal(
+                FeatureTransforms.apply_append(x, T; inds=[1, 2], append_dim=2),
+                hcat(x, ones(2, 2))
+            )
+        end
+
+        @testset "ManyToOne" begin
+
+            T = FakeManyToOneTransform()
+
+            @test FeatureTransforms.apply_append(x, T; dims=1, append_dim=1) == vcat(x, ones(1, 3))
+            @test FeatureTransforms.apply_append(x, T; dims=2, append_dim=2) == hcat(x, ones(2))
+        end
+
+        @testset "ManyToMany" begin
+
+            T = FakeManyToManyTransform()
+
+            @test_throws DimensionMismatch FeatureTransforms.apply_append(x, T; append_dim=1)
+            @test_throws DimensionMismatch FeatureTransforms.apply_append(x, T; append_dim=3)
+
+            @test isequal(
+                FeatureTransforms.apply_append(x, T; append_dim=2),
+                hcat(x, ones(2, 6))
+            )
+
+            @test isequal(
+                FeatureTransforms.apply_append(x, T; inds=[1, 2], append_dim=2),
+                hcat(x, ones(2, 2))
+            )
+        end
+
+    end
+
+    if ArrayType == KeyedArray
```

Comment:
I forget, are there issues with AxisArray here? Or are you just not bothering?

Reply:
The dims syntax for AxisArrays expects an `Axis` type not a `Symbol` so it's incompatible with our interface unless we call `axisdim(A, Axis{:foo})` or something to extract the dimension index.

Something we could do maybe to support this without depending on AxisArrays is:
```julia
if hasfield(A, axes)
    dims = only(findall(a -> a == dims || typeof(a) <: dims, A.axes))
end
```

but I'd prefer to implement that at a later time if we actually need it (and I'd almost prefer to drop AxisArrays than do that).